File: backend/src/domain/entities/MentorAvailability.ts
import { Entity, PrimaryGeneratedColumn, Column } from 'typeorm';

@Entity('mentor_availabilities')
export class MentorAvailability {
  @PrimaryGeneratedColumn()
  id: number;

  @Column({ name: 'mentor_member_id' })
  mentorMemberId: number;

  @Column({ type: 'timestamp', name: 'start_time', nullable: true })
  startTime: Date;

  @Column({ type: 'timestamp', name: 'end_time', nullable: true })
  endTime: Date;

  @Column({ type: 'varchar', default: 'Available' })
  status: string;
}

File: backend_new_structure/common/src/main/java/com/service/common/entity/MentorAvailability.java
package com.service.common.entity;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.springframework.data.annotation.Id;
import org.springframework.data.relational.core.mapping.Column;
import org.springframework.data.relational.core.mapping.Table;

import java.time.LocalDateTime;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
@Table("mentor_availabilities")
public class MentorAvailability {

    @Id
    private Long id;

    @Column("mentor_member_id")
    private Long mentorMemberId;

    @Column("start_time")
    private LocalDateTime startTime;

    @Column("end_time")
    private LocalDateTime endTime;

    @Builder.Default
    private String status = "Available";
}
